Observer/NonBoltOrderObserver.php

```diff
@@ -0,0 +1,232 @@
+            }
+        } catch (Exception $exception) {
+            $this->metricsClient->processCountMetric("non_bolt_order_creation.failure", 1);
+            $this->bugsnag->notifyException($exception);
+            return;
+        } catch (Error $error) {
+            // catch errors so failures here don't prevent orders from being created
+            $this->metricsClient->processCountMetric("non_bolt_order_creation.failure", 1);
+            $this->bugsnag->notifyException($error);
+            return;
+        }
+
+        $this->metricsClient->processCountMetric("non_bolt_order_creation.success", 1);
+    }
+
+    /**
+     * Call Bolt Create Non-Bolt Order API
+     *
+     * @param array $cart
+     * @param CustomerInterface $customer
+     * @param int $storeId
+     * @return Response|int
+     * @throws LocalizedException
+     * @throws Zend_Http_Client_Exception
+     */
+    protected function createNonBoltOrder($cart, $customer, $storeId)
+    {
+        $apiKey = $this->configHelper->getApiKey($storeId);
+
+        $phone = $this->getPhone($cart);
+        $requestData = $this->dataObjectFactory->create();
+        $requestData->setApiData([
+            'cart' => $cart,
+            'user_identifier' => [
+                'email' => $customer->getEmail(),
+                'phone' => $phone,
+            ],
+            'user_identity' => [
+                'first_name' => $customer->getFirstname(),
+                'last_name' => $customer->getLastname(),
+            ],
+        ]);
+        $requestData->setDynamicApiUrl(ApiHelper::API_CREATE_NON_BOLT_ORDER);
+        $requestData->setApiKey($apiKey);
+        $requestData->setStatusOnly(true);
+
+        $request = $this->apiHelper->buildRequest($requestData);
+        return $this->apiHelper->sendRequest($request);
+    }
+
+    /**
+     * Get phone number from the cart if it exists
+     *
+     * @param array $cart
+     * @return string
+     */
+    protected function getPhone($cart)
+    {
+        if (!@$cart['shipments']) {
+            return null;
+        }
+
+        if (count($cart['shipments']) < 1) {
+            return null;
+        }
+
+        if (!@$cart['shipments'][0]['shipping_address']) {
+            return null;
+        }
+
+        return $cart['shipments'][0]['shipping_address']['phone'];
+    }
+
+    /**
+     * Workaround/hack for PayPal orders that might have first and last name in the address's FirstName attribute
+     *
+     * @param $quote
+     */
+    protected function handleMissingName($quote) {
+        $address = $quote->isVirtual() ? $quote->getBillingAddress() : $quote->getShippingAddress();
+        if ($address->getLastName() == null) {
+            $name = $address->getFirstName();
+            $names = preg_split("/[\s]+/", $name);
+            if (count($names) > 1) {
+                $address->setFirstName($names[0]);
+                $address->setLastName($names[1]);
+            } else if (count($names) > 2) {
```

Comment:
If the previous condition, `if (count($names) > 1)`, is fulfilled you never get here.
The new block below should replace the old block above cause it covers both (all) cases.

Reply:
see below